spec/requests/legacy/redirect_spec.rb

```diff
@@ -38,6 +38,63 @@
     end
   end
 
+  describe 'datafile resources' do
+    let(:legacy_dataset_name) { 'a-legacy-name' }
+    let(:datafile) { CSV_DATAFILE.with_indifferent_access }
+
+    let(:dataset) do
+      DatasetBuilder
+        .new
+        .with_legacy_name(legacy_dataset_name)
+        .with_datafiles([datafile])
+        .build
+    end
+
+    context 'when the datafile can not be found' do
+      it '' do
+        get "/dataset/#{legacy_dataset_name}/resource/#{datafile[:uuid]}"
+
+        expect(response).to have_http_status(:not_found)
+      end
+    end
+
+    context 'when the datafile can not be previewed' do
+      before do
+        stub_request(:get, datafile[:url]).to_return(status: [404, "Not Found"])
+        index([dataset])
+      end
+
+      it 'redirects to the files part of the dataset page' do
```

Comment:
I'm not sure about that redirect. Even if the dataset can't be previewed, the preview page exists and not only says 'can't be previewed' but also offers a download button.

Reply:
Okay, that's a more elegant solution. I'll do that!

Reply:
Fixed with https://github.com/alphagov/datagovuk_find/pull/393/commits/5b5639973b9c6b70e7c509c175bb49f79f8b6e74